ixbrlparse/components/nonnumeric.py

```diff
@@ -21,12 +24,32 @@ def __init__(
             self.schema = "unknown"
             self.name = name_split[0]
 
+        self.text = value
         self.context = context
         self.format = format_
         self.value = value
 
+        format_ = {
+            "format_": self.format,
+            "numeric": False,
+            "decimals": None,
+            "scale": None,
+            "sign": None,
```

Comment:
I think we should be able to leave these out here, might be clearer (although to help we would need to set a default of `None` for `decimals` in `ixbrlFormat`.